frontend/src/components/Seller/SellerListingForm.tsx

```diff
@@ -19,36 +24,61 @@ interface IFormInput {
   inventory: number;
   price: number;
   releaseDate: string;
+  listingId?: string;
 }
 
 export default function SellerListingForm(props: ISellerListingFormProps) {
   const defaultValues = {
     sellerUserId: props.sellerId,
-    isbn: "",
-    title: "",
-    author: "",
-    publisher: "",
-    description: "",
-    inventory: 0,
-    price: 0.0,
+    isbn:props.formValues && props.isEdit ?  props.formValues?.isbn : "",
+    title:props.formValues && props.isEdit ? props.formValues?.title : "",
+    author:props.formValues && props.isEdit ? props.formValues?.author : "",
+    publisher:props.formValues && props.isEdit ? props.formValues?.publisher : "",
+    description:props.formValues && props.isEdit ? props.formValues?.description : "",
+    inventory:props.formValues && props.isEdit ? props.formValues?.inventory : 0,
+    price:props.formValues && props.isEdit ? props.formValues?.price : 0.0,
     releaseDate: new Date().toISOString().slice(0, 10),
   };
+
   const { sendRequest } = useHttpClient();
   const formMethods = useForm({ defaultValues });
   const {
     handleSubmit,
     control,
     formState: { errors },
   } = formMethods;
+
+
   const onSubmit: SubmitHandler<IFormInput> = async (data) => {
-    const dataCopy = JSON.parse(JSON.stringify(data));
-    dataCopy.sellerUserId = parseInt(dataCopy.sellerUserId);
-    dataCopy.price = parseFloat(dataCopy.price).toFixed(2);
-    dataCopy.inventory = parseInt(dataCopy.inventory);
-    dataCopy.coverImage = "/static/images/book-cover.jpg";
-    await sendRequest(LISTING, "POST", dataCopy);
-    props.handleCloseDialog();
+      if(props.isEdit)
+      {
+          const dataCopy = JSON.parse(JSON.stringify(data));
```

Comment:
The dataCopy is just being copied. 

Maybe put the common code outside the If

Reply:
Copy moved outside if